### PR TITLE
Language.py: Raise AttributeError in __getattr__ 

### PR DESCRIPTION
**Pull request by hemangsk**
### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
- [x] I have followed the [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference).
- [ ] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### Reviewers

Fixes https://github.com/coala/coala/issues/3160

### CODE REVIEW

**Comment by Mixih** [comment]
No brackets around the "WIP". Gitmate doe snto understand those.

**Comment by Mixih** [comment]
All reviews on hold because of circle acting up. Please be patient as they work to fix their problems :)

**Comment by sils** [comment]
remove colon from commit message

File: coalib/bearlib/languages/Language.py [hunk 1]
```diff
@@ -339,7 +339,10 @@ def __getattr__(self, item):
         if len(self.versions) > 1:
             raise AttributeError('You have to specify ONE version of your '
                                  'language to retrieve attributes for it.')
-        return self._attributes[item]
+        try:
+            return self._attributes[item]
+        except KeyError:
+            raise AttributeError
```
Comment (sils) [inline coalib/bearlib/languages/Language.py]: maybe give it a nice message listing the avail. attributes?

File: coalib/bearlib/languages/Language.py [hunk 2]
```diff
@@ -339,7 +339,17 @@ def __getattr__(self, item):
...
-        return self._attributes[item]
+        try:
+            return self._attributes[item]
+        except KeyError:
+            if len(self.attributes) == 0:
+                message = 'There are no available attributes for this language.'
+            else:
+                message = 'This is not a valid attribute! '
+                '\nThe following attributes are available:-'
+                for attr in self.attributes:
```
Comment (jayvdb) [inline coalib/bearlib/languages/Language.py]: Use `'\n'.join(self.attributes)`

File: tests/bearlib/languages/LanguageTest.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    def tearDown(self):
+        pass
+
+    def test_invalid_attribute(self):
+        with self.assertRaises(AttributeError):
+            self.lang_cpp.not_an_attribute
+
+    def test_attribute_list_empy(self):
+        with self.assertRaises(AttributeError):
```
Comment (jayvdb) [inline tests/bearlib/languages/LanguageTest.py]: Use assertRaisesRegexp with a short part of the message to verify you have the correct message

File: coalib/bearlib/languages/Language.py [hunk 4]
```diff
@@ -339,7 +339,16 @@ def __getattr__(self, item):
...
                                  'language to retrieve attributes for it.')
-        return self._attributes[item]
+        try:
+            return self._attributes[item]
+        except KeyError:
+            if len(self.attributes) == 0:
+                message = 'There are no available attributes for this language.'
+            else:
+                message = ('This is not a valid attribute! '
+                           '\nThe following attributes are available-')
```
Comment (sils) [inline coalib/bearlib/languages/Language.py]: ah no why does this with a `-`?

**Comment by sils** [comment]
unack 3f0f80b

File: coalib/bearlib/languages/Language.py (hunk 4 above)
Comment (hemangsk) [inline coalib/bearlib/languages/Language.py]: oops!  reverting it back  `:-`

File: coalib/bearlib/languages/Language.py [hunk 5]
```diff
@@ -339,7 +339,16 @@ def __getattr__(self, item):
...
                                  'language to retrieve attributes for it.')
-        return self._attributes[item]
+        try:
+            return self._attributes[item]
+        except KeyError:
+            if len(self.attributes) == 0:
+                message = 'There are no available attributes for this language.'
+            else:
+                message = ('This is not a valid attribute! '
+                           '\nThe following attributes are available:-')
```
Comment (sils) [inline coalib/bearlib/languages/Language.py]: why's ther ea minus!?

Comment (hemangsk) [inline coalib/bearlib/languages/Language.py]: ohh sorry removing it. i thought it was about the missing colon with minus

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3168/commits/54c926112b40bca11f51067144c7e5b3031534ca

**Comment by Mixih** [comment]
@jayvdb Time for reevaluation

**Comment by jayvdb** [comment]
has conflicts; needs rebasing.

**Comment by Mixih** [comment]
@hemangsk Please rebase and resolve the conflict.

**Comment by sils** [comment]
reack https://github.com/coala/coala/pull/3168/commits/dcbb31be98f75cf39c2cdf5eaad342ebdb809b95

**Comment by sils** [comment]
finally!

**Comment by sils** [comment]
@rultor merge

**Comment by sils** [comment]
reack dcbb31b

**Pull request merged**